web/src/settings_account.ts

```diff
@@ -279,6 +282,53 @@ export function update_privacy_settings_box(property: PrivacySettingName): void
     settings_components.set_input_element_value($input_elem, user_settings[property]);
 }
 
+export function render_user_timezone_dropdown_widget(): void {
+    const timezone_items = timezones.timezones.map((tz: {name: string; utc_offset: string}) => ({
+        name: `${tz.name} (${tz.utc_offset})`,
+        unique_id: tz.name,
+    }));
+
+    const opts: dropdown_widget.DropdownWidgetOptions = {
+        widget_name: "user_timezone",
+        get_options: () => timezone_items,
+        item_click_callback(event, dropdown, widget) {
+            dropdown.hide();
+            event.preventDefault();
+            event.stopPropagation();
+            const selected = widget.value();
+            settings_ui.do_settings_change(
+                channel.patch,
+                "/json/settings",
+                {timezone: selected},
+                $(".timezone-setting-status").expectOne(),
+                {
+                    success_continuation() {
+                        /*
+                            When user selects the option, update the widget to show the selected option.
+                            */
+                        const prev = user_settings.timezone;
+                        try {
+                            widget.render(selected);
+                        } catch {
+                            widget.render(prev);
```

Comment:
Here I meant to just update the widget to show selected option witout waiting for request to succeed and then updating the value to previous value if request returns an error.